e2e-test/java/io/cdap/plugin/slt/stepsdesign/StepDefinitionSLT.java

```diff
@@ -0,0 +1,471 @@
+
+/**
+ * Contains SLT test scenarios step definitions.
+ */
+public class StepDefinitionSLT implements CdfHelper {
+
+  private static final Logger logger = LoggerFactory.getLogger(PluginPropertyUtils.class);
+  private SAPProperties sapProps;
+  private ErrorCapture errorCapture;
+  private SAPAdapterImpl sapAdapterImpl;
+  static String action;
+  private ExceptionUtils exceptionUtils;
+  public static SLTActions sltActions = new SLTActions();
+  public static SLTLocators sltLocators = new SLTLocators();
+  public static String rawLog = null;
+  BigQueryClient bqClient = new BigQueryClient();
+  static int sapRecordsCount;
+  static int bqRecordCount = 0;
+  private static ArrayList<String> tabs = null;
+  private static String parent = null;
+  private static Properties connection = new SAPProperties();
+  private List<String> fields = new ArrayList<>();
+  private static boolean isErrorPresent = false;
+  private static String mtId = null;
+  private static String guid = null;
+  static boolean errorExist = false;
+  static String color;
+  private static AssertionHelper assertionHelper = new AssertionHelper();
+  private static ElementHelper elementHelper = new ElementHelper();
+  private static WaitHelper waitHelper = new WaitHelper();
+  private static PluginPropertyUtils pluginPropertyUtils = new PluginPropertyUtils();
+
+
+  static {
+    sltLocators = SeleniumHelper.getPropertiesLocators(SLTLocators.class);
+    Properties prop = new Properties();
+    try {
+      prop.load(new FileInputStream("src/e2e-test/resources/Google_SAP_Connection.properties"));
+    } catch (IOException e) {
+      logger.error("Failed while reading SAP properties file" + e);
+    }
+    for (String property : prop.stringPropertyNames()) {
+      connection.put(property, prop.getProperty(property));
+    }
+  }
+
+  @Then("^Set Dummy MTID")
+  public void setDummyMTID() {
+    mtId = "02K";
+  }
+
+
+  @Given("^Open CDF replication and initiate pipeline creation")
+  public void open_cdf_replication() throws IOException, InterruptedException {
+    openCdf();
+    SeleniumDriver.getDriver().get("http://localhost:11011/cdap/ns/default/replication/create");
+    sltActions.enterName(UUID.randomUUID().toString().replaceAll("-", ""));
+    sltActions.clickNextButton();
+    isErrorPresent = false;
+    bqRecordCount = 0;
+    sapRecordsCount = 0;
+  }
+
+  @When("Select SAP SLT as source and fill connection parameters")
+  public void source_is_sap_slt() {
+    sltActions.clickSltPlugin();
+    sltActions.enterMandatoryParameters(
+      "gcpProjectId",
+      "gcsDataPath",
+      mtId,
+      "sapJcoLibGcsPath",
+      "jcoclienthost",
+      "jcoclient",
+      "jcoclientsysnr",
+      "jcoclientuser",
+      "jcoclientpasswd");
+  }
+
+  @When("Select SAP SLT as source")
+  public void source_is_sapslt() {
+    sltActions.clickSltPlugin();
+  }
+
+  @Then("Verify that after setting SLT parameter {string} as {string} plugin is throws error {string}")
```

Comment:
Separate steps:
1. Enter SLT parameter {string} with value {string}
2. Verify SLT parameter validation error message {string} is displayed